hnn_core/network.py

```diff
@@ -310,10 +312,27 @@ def _instantiate_feeds(self, n_trials=1):
                                 params=p_unique[src_type],
                                 gid=gid)
                         )
-                event_times_per_source.update({src_type: event_times})
-
-            # list of dict of list of list
-            self.trial_event_times.append(event_times_per_source.copy())
+                feed_times[src_type].append(event_times)
+
+        # Add tonic inputs if present
+        feed_times['tonic'] = dict()
+        for cell_type in ['L2Basket', 'L5Basket', 'L2Pyr', 'L5Pyr']:
+            is_tonic_present = [f'Itonic_{p}_{cell_type}_soma' in self.params
+                                for p in ['A', 't0', 'T']]
+            if any(is_tonic_present):
+                if not all(is_tonic_present):
+                    raise ValueError(f'Tonic input must have the amplitude, '
+                                     f'start time and end time specified. One '
+                                     f'or more parameter may be missing for '
+                                     f'cell type {cell_type}')
+                feed_times['tonic'][cell_type] = {
+                    'amplitude': self.params[f'Itonic_A_{cell_type}_soma'],
+                    't0': self.params[f'Itonic_t0_{cell_type}_soma'],
+                    'T': self.params[f'Itonic_T_{cell_type}_soma']
+                }
+
+        # list of dict of list of list
```

Comment:
Is this a mistake? Seems to disagree with DOC on L164:
```
feed_times: dict of list (n_trials) of list (n_cells) of list (n_times)
```

Reply:
cruft from old code :)